app/src/main/java/com/asfoundation/wallet/logging/send_logs/SendLogsRepository.kt

```diff
@@ -0,0 +1,134 @@
+package com.asfoundation.wallet.logging.send_logs
+
+import com.asfoundation.wallet.base.RxSchedulers
+import io.reactivex.Completable
+import io.reactivex.Maybe
+import io.reactivex.Observable
+import io.reactivex.Single
+import io.reactivex.subjects.BehaviorSubject
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.MultipartBody
+import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.asRequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import retrofit2.http.*
+import java.io.File
+
+class SendLogsRepository(
+    private val sendLogsApi: SendLogsApi,
+    private val awsUploadFilesApi: AwsUploadFilesApi,
+    private val logsDao: LogsDao,
+    private val rxSchedulers: RxSchedulers,
+    private val cacheDir: File,
+) {
+
+  private val sendStateBehaviorSubject = BehaviorSubject.createDefault(SendState.UNINITIALIZED)
+
+  fun saveLog(tag: String?, data: String): Completable {
+    return Completable.fromAction { logsDao.saveLog(LogEntity(null, tag = tag, data = data)) }
+        .subscribeOn(rxSchedulers.io)
+  }
+
+  fun updateLogs(): Maybe<File> {
+
+    return logsDao.updateLogs()
+        .andThen(
+            logsDao.getSendingLogs()
+                .flatMapMaybe { logs ->
+                  if(logs.isEmpty())
+                    return@flatMapMaybe Maybe.empty()
+
+                  val logsFile = File.createTempFile("log", null, cacheDir)
+                  val logContent = StringBuilder()
+
+                  logs.forEach { logEntity ->
+                    logContent.append(logEntity.created.toString())
+                    logEntity.tag?.let { logContent.append(" " + logEntity.tag) }
+                    logContent.appendLine(": " + logEntity.data)
+
+                    logsFile.appendText(logContent.toString())
+                    logContent.clear()
+                  }
```

Comment:
This is minor and should be left to the end if we have time, but maybe we could skip creating the file!

So, it seems there's an extension method File.asRequestBody("text/*".toMediaType()) that we're using. This creates a RequestBody, that consequently uses a BufferedSink to write info into it. Maybe we could just use this directly and put our logs directly here. This way we skip a lot of I/O overhead by creating and reading the file.

So ideally, instead of returning a File, we could return the RequestBody directly.